### PR TITLE
[sociald] Use Facebook API v2.2

### PR DESCRIPTION
**Pull request by antseppa**
Upgraded to use Facebook graph API 2.2

Biggest changes:
  - graph API base address and version now comes from account setting
  - changed calendar sync to use batch query instead of FQL (and added handling for paged requests)
  - removed postsyncadaptor as it was using FQL and there's no need for it now

### CODE REVIEW

File: src/facebook/facebook-calendars/facebookcalendarsyncadaptor.cpp [hunk 1]
```diff
@@ -38,6 +38,23 @@
 static const char *FACEBOOK = "Facebook";
 static const char *FACEBOOK_COLOR = "#3B5998";
 
+inline FacebookParsedEvent::FacebookParsedEvent()
```
Comment (SfietKonstantin) [inline src/facebook/facebook-calendars/facebookcalendarsyncadaptor.cpp]: No need to inline code. Compiler will ignore it it seems

File: src/facebook/facebookdatatypesyncadaptor.h [hunk 2]
```diff
@@ -54,6 +54,7 @@ class FacebookDataTypeSyncAdaptor : public SocialNetworkSyncAdaptor
 
 protected:
     QString clientId();
+    QString graphAPI() const;
```
Comment (SfietKonstantin) [inline src/facebook/facebookdatatypesyncadaptor.h]: What about graphApi(const QString &request) ?
Used like graphApi("me")

File: src/facebook/facebookdatatypesyncadaptor.h [hunk 3]
```diff
@@ -71,6 +72,10 @@ private Q_SLOTS:
     void signIn(Accounts::Account *account);
     bool m_triedLoading; // Is true if we tried to load (even if we failed)
     QString m_clientId;
+    QString m_graphAPI;
 };
 
+inline QString FacebookDataTypeSyncAdaptor::graphAPI() const
+{ return m_graphAPI; }
```
Comment (chriadam) [inline src/facebook/facebookdatatypesyncadaptor.h]: I'd prefer if the implementation was put into the .cpp unless we're 100% certain that the implementation of this function won't change in the future.

File: src/facebook/facebookdatatypesyncadaptor.cpp [hunk 4]
```diff
@@ -252,6 +252,8 @@ void FacebookDataTypeSyncAdaptor::signOnResponse(const SignOn::SessionData &resp
         SOCIALD_LOG_INFO("signon response for account with id" << accountId << "contained no access token");
     }
 
+    m_graphAPI = account->value(QStringLiteral("graph_api/Host")).toString();
```
Comment (chriadam) [inline src/facebook/facebookdatatypesyncadaptor.cpp]: will this value be different per-account, or should this be a service template setting?

Comment (chriadam) [inline src/facebook/facebookdatatypesyncadaptor.cpp]: Uh, ignore me.  I forgot that Accounts&SSO doesn't allow retrieving template settings without instantiating the account instance.

Comment (chriadam) [inline src/facebook/facebookdatatypesyncadaptor.cpp]: You may, however, need to select an appropriate service, unless it's also a global-service setting?

File: src/facebook/facebookdatatypesyncadaptor.h (hunk 3 above)
Comment (antseppa) [inline src/facebook/facebookdatatypesyncadaptor.h]: Better safe than sorry, I'll change it.

File: src/facebook/facebookdatatypesyncadaptor.cpp (hunk 4 above)
Comment (antseppa) [inline src/facebook/facebookdatatypesyncadaptor.cpp]: It is also a global service setting (defined in .provider -file).

**Comment by antseppa** [comment]
Updated: removed inlining and made graphApi() method slightly cleaner to use.

**Comment by chriadam** [comment]
LGTM

**Pull request merged**